api/app/persistence/Store.scala

```diff
@@ -0,0 +1,263 @@
+package persistence
+
+import org.apache.http.HttpHost
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.action.get.{GetRequest, GetResponse}
+import org.elasticsearch.action.index.{IndexRequest, IndexResponse}
+import org.elasticsearch.action.search.{SearchRequest, SearchResponse}
+import org.elasticsearch.client.{
+  RequestOptions,
+  RestClient,
+  RestHighLevelClient
+}
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.index.query.QueryBuilders
+import org.elasticsearch.search.builder.SearchSourceBuilder
+import play.api.inject.ApplicationLifecycle
+import play.api.libs.json._
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.{Failure, Success}
+
+import models.{Slot, Test, Variant}
+
+trait Store {
+  def getAllSlots(): Future[List[Slot]]
+  def getSlot(id: String): Future[Option[Slot]]
+  def updateSlot(slotId: String, updatedSlot: Slot): Future[Unit]
+  def getAllVariants(): Future[List[Variant]]
```

Comment:
This is the bit I'd pull into a separate trait and then mix that into Store. We can then use that trait for any controllers/tests that only need variant info.

    trait VariantStore {
      def getAllVariants() ...
    }

    trait SlotStore {
      .. // slot methods/interface
    }

    trait Store extends VariantStore with SlotStore

or something like that?

The motivation is to keep the interface narrow for controllers/tests that only care about one.

Reply:
Ah cool yeah that makes sense, will update the PR :+1: